### DISCUSSION TITLE
How to use Pinia in *.js env. without VUE Cli ?

### DISCUSSION THREAD

**Discussion by freatraum** [vuejs/pinia#2280] [Help and Questions]
I want to use Pinia in .js with out VUE Cli. 
This is my files:
index.html
index.js
index.vue

**Comment by roma-marshall** [comment]
To use Pinia in a JavaScript project without Vue CLI, you can follow these steps:

1. Create a new directory for your project.

2. Inside the project directory, create the following files:

- **index.html**
- **index.js**
- **index.vue**

3. Open index.html and add the following code:
```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Pinia Example</title>
</head>
<body>
  <div id="app"></div>
  <script src="index.js"></script>
</body>
</html>
```

4. Open index.js and add the following code:
```
import { createApp } from 'vue';
import { createPinia } from 'pinia';
import App from './index.vue';

const app = createApp(App);
const pinia = createPinia();

app.use(pinia);
app.mount('#app');
```

5. Open index.vue and add the following code:
```
<template>
  <div>
    <h1>Hello, Pinia!</h1>
  </div>
</template>

<script>
export default {
  name: 'App',
}
</script>

<style>
h1 {
  color: blue;
}
</style>
```

6. Now, open a terminal, navigate to your project directory, and install the required dependencies by running:
```
npm init -y
npm install vue@next @pinia/next
```

7. Once the dependencies are installed, you can start a local development server by installing a simple server package, such as 
http-server. Run the following command in your terminal:
```
npm install -g http-server
http-server
```

8. Open your web browser and navigate to http://localhost:8080

**Reply by freatraum** [reply]
Thank you for your reply. But it's still not quite what I wanted. It may be that I expressed it not clearly. I mean, only use Pinia and Vue in the form of importing js files. Now I have Vue.js but no Pinia .js. Is there any way to compile a Pinia .js?
